src/Aspire.Hosting/ApplicationModel/ExpressionResolver.cs

```diff
@@ -148,22 +151,28 @@ async Task<ResolvedValue> EvalValueProvider(IValueProvider vp)
         return new ResolvedValue(value, false);
     }
 
+    async Task<ResolvedValue> ResolveConnectionStringReferenceAsync(ConnectionStringReference cs)
+    {
+        // GetValueAsync will throw if the connection string is not optional and does not exist.
+        // Invoke GetValueAsync to make sure the value isn't resolved, since we are manually resolving the expression.
```

Comment:
Can we beef up this comment a little bit, so it is more clear why we are calling GetValueAsync? Is it only to produce the exception and not duplicate code?